### PR TITLE
platform specific env setup

### PR DESCRIPTION
**Pull request by gregw**
This is a bit of a thought bubble....  however I think it would greatly simplify extending the environment setup if we had:
 a) a directory based mechanism so that scripts could be added rather than appended; 
 b) a platform specific script that would select other directories of scripts based on a single platform selector

This PR achieves this, however the platform selection algorithm  for local, gae, aws and heroku is rather rudimentary and needs to be extended to support additional flavours of GCP.

However, with a mechanism like this, it would be much easier to create images that will run anywhere without each mechanism repeating the platform detection heuristics.

Currently only the cloud debugger is in a platform specific location, however the jetty-runtime image would also benefit by dropping in the configuration for its `gae` module into only the `platform-gae.d` directory.

### CODE REVIEW

**Comment by meltsufin** [comment]
This is an interesting idea, but I don't think we're ready to put this kind of structure in place before we have a better idea of what will go into each of these environment files. For the time being, I would rather keep it simple, though less "organized".  See: #74. We should later do a refactoring like what you're suggesting here.

**Comment by gregw** [comment]
@meltsufin you are right that putting in a complex generic structure based on an example of 1 is overkill for now.  However, I am still drawn to the simplicity and flexibility of a `setup-env.d` that is added to or deleted from rather than a `setup-env.bash` that is appended to or replaced.

So I've updated to a halfway solution between what was originally proposed and your #74. What do you think?

**Comment by meltsufin** [comment]
@gregw The halfway solution might work. Did you forget to push the change though?

**Comment by gregw** [comment]
@meltsufin oops sorry. Pushed now.

File: openjdk8/src/main/docker/setup-env.d/10-setup-env.bash [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+isTrue() {
```
Comment (meltsufin) [inline openjdk8/src/main/docker/setup-env.d/10-setup-env.bash]: Let's get rid of this method.

File: openjdk8/src/main/docker/setup-env.d/10-setup-env.bash [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (meltsufin) [inline openjdk8/src/main/docker/setup-env.d/10-setup-env.bash]: I would rename this file to `10-platform-env.bash`

File: openjdk8/src/main/docker/docker-entrypoint.bash [hunk 3]
```diff
@@ -1,8 +1,12 @@
 #!/bin/bash
 
 # source the supported feature JVM arguments
```
Comment (meltsufin) [inline openjdk8/src/main/docker/docker-entrypoint.bash]: Change comment to reflect the block bellow

File: openjdk8/src/test/resources/structure.yaml [hunk 4]
```diff
@@ -35,9 +35,9 @@ fileExistenceTests:
   path: '/docker-entrypoint.bash'
   isDirectory: false
   shouldExist: true
-- name: 'Env setup script exists'
-  path: '/setup-env.bash'
-  isDirectory: false
+- name: 'Env setup scripts exists'
```
Comment (meltsufin) [inline openjdk8/src/test/resources/structure.yaml]: Let's add the new files in here for testing as well.

**Comment by gregw** [comment]
@meltsufin Great! when will this be available as a build from gcr.io ?

**Comment by meltsufin** [comment]
@aslo Do you want to try cutting a release using the new pipeline?

**Comment by aslo** [comment]
Sure, I'll give it a go today.

**Comment by meltsufin** [comment]
@gregw  It's available now: `gcr.io/google-appengine/openjdk:8-2017-04-12_14_02`

**Pull request merged**